presto-main/src/main/java/io/prestosql/operator/scalar/DateTimeFunctions.java

```diff
@@ -982,7 +982,11 @@ public static long dayFromInterval(@SqlType(StandardTypes.INTERVAL_DAY_TO_SECOND
     public static long lastDayOfMonthFromTimestampWithTimeZone(@SqlType(StandardTypes.TIMESTAMP_WITH_TIME_ZONE) long timestampWithTimeZone)
     {
         long millis = unpackMillisUtc(timestampWithTimeZone);
-        millis = unpackChronology(timestampWithTimeZone).monthOfYear().roundCeiling(millis + 1);
+        ISOChronology isoChronology = unpackChronology(timestampWithTimeZone);
+        millis = isoChronology.monthOfYear().roundCeiling(millis + 1);
+        // millis is currently midnight in timezone of the original value
```

Comment:
This comment makes me think about the case when there is no midnight and the date/time is near the end of the month.

For example, `America/Bahia_Banderas` and the local date/time 1970-01-01 00:00:00 ±1ms.

Can you add this to your tests? (along with other zones and date/times you have)

Reply:
Added tests for timezone `America/Bahia_Banderas` and `2019-01-01 00:00:00.000 ±1ms`.
Not sure how to test for day without midnight. Could you provide a concrete example for this?
`1970-01-01 00:00:00` is epoch and difficult to test.